docs/guide/widgets.md

```diff
@@ -51,6 +51,32 @@ The addition of the CSS has completely transformed our custom widget.
 ```{.textual path="docs/examples/guide/widgets/hello02.py"}
 ```
 
+### Responding to key presses
```

Comment:
I think we need a more elaborate example here.

We've managed to avoid abbreviated code. Virtually everything has a fully working example for any given feature.

Suggest we come up with some kind of input widget that shows focus, and does *something*. How about a widget that displays a 24-hour clock time. Where clicking on a digit increments that digit. Just an idea.

Reply:
Maybe I'm misunderstanding but I don't think that idea would really demonstrate bindings or focus? I find examples with that kind of scope to be a major distraction too - 90% of the code is boiler plate and the actual stuff related to the concept we're trying to explain is buried away. I personally end up skipping examples like that due to the verbosity (and my goldfish-like attention span)!

I have a simple keyboard driven `Counter` widget I'm using to test keymaps that might make a good example? You focus it and then use up/right/k/plus to increment, and can decrement too.